event_types.go

```diff
@@ -71,26 +71,52 @@ func (event Event) String() string {
 }
 
 // merge combines this event with the other event (assumed to be newer!).
-// TODO: Intelligent merging: if there are multiple EventUpdate(Flags) events, can we just take the latest one?
-func (event *Event) merge(other Event) error {
-	event.EventID = other.EventID
+func (event *Event) merge(next Event) error {
+	event.EventID = next.EventID
 
-	if other.User != nil {
-		event.User = other.User
+	if next.User != nil {
+		event.User = next.User
 	}
 
-	if other.MailSettings != nil {
-		event.MailSettings = other.MailSettings
+	if next.MailSettings != nil {
+		event.MailSettings = next.MailSettings
 	}
 
-	// For now, label events are simply appended.
-	event.Labels = append(event.Labels, other.Labels...)
+	for _, nextLabel := range next.Labels {
```

Comment:
Do we have any "proof" that is a better approach?

Reply:
Yes -- when there are a huge number of events, the API will use `More` to indicate that there are more events needing to be fetched, but it won't compact them because there are too many. So we can do the compacting ourselves to be more efficient.